commit 2c0e40bfbb3ead26353861e3566c653085e977a3
Author: nick <nick@a0aa69c2-05f4-0310-b83c-d5d913b14636>
Date:   Wed Mar 11 18:00:30 2009 +0000

    Remove unsigned clamp macro that broke applying the display settings (bug #4966).
    
    
    git-svn-id: http://svn.xfce.org/svn/xfce/xfce4-settings/trunk@29650 a0aa69c2-05f4-0310-b83c-d5d913b14636

diff --git a/dialogs/display-settings/xfce-randr-legacy.c b/dialogs/display-settings/xfce-randr-legacy.c
index 53b272e..95bc823 100644
--- a/dialogs/display-settings/xfce-randr-legacy.c
+++ b/dialogs/display-settings/xfce-randr-legacy.c
@@ -27,11 +27,6 @@
 
 #include "xfce-randr-legacy.h"
 
-/** CLAMP gives a warning with comparing unsigned values */
-#define UCLAMP(x, low, high) \
-  (((x) > (high)) ? (high) : \
-   (((low) > 0) ? (((low) < (x)) ? (low) : (x)) : 0))
-
 
 
 XfceRandrLegacy *
@@ -184,7 +179,7 @@ xfce_randr_legacy_save (XfceRandrLegacy *legacy,
     {
         /* find the resolution and save it as a string */
         sizes = XRRConfigSizes (legacy->config[n], &nsizes);
-        size = UCLAMP (legacy->resolution[n], 0, nsizes - 1);
+        size = MIN (legacy->resolution[n], MAX (0, nsizes - 1));
         resolution = g_strdup_printf ("%dx%d", sizes[size].width, sizes[size].height);
         g_snprintf (property, sizeof (property), "/%s/Screen_%d/Resolution", scheme, n);
         xfconf_channel_set_string (channel, property, resolution);
